src/map/pc.cpp

```diff
@@ -12051,15 +12055,17 @@ static void pc_unequipitem_sub(map_session_data *sd, int n, int flag) {
 				if (!sd->inventory.u.items_inventory[n].card[i])
 					continue;
 
+				current_equip_card_id = sd->inventory.u.items_inventory[n].card[i];
 				std::shared_ptr<item_data> data = item_db.find(sd->inventory.u.items_inventory[n].card[i]);
 
 				if (data != nullptr) {
 					if (data->unequip_script)
 						run_script(data->unequip_script, 0, sd->bl.id, fake_nd->bl.id);
```

Comment:
I would prefer to have the current_equip_card_id just around this run_script call.